### DISCUSSION TITLE
Does Datafusion support manually specifying the timestamp and time precision?

### DISCUSSION THREAD

**Discussion by caicancai** [apache/datafusion#13230] [Q&A] [answered]
Here is a quote from the postgres documentation: https://www.postgresql.org/docs/16/datatype-datetime.html

time, timestamp, and interval accept an optional precision value p which specifies the number of fractional digits retained in the seconds field. By default, there is no explicit bound on precision. The allowed range of p is from 0 to 6.

**Comment by findepi** [comment, marked as answer]
On SQL frontend layer, only `timestamp` is currently supported and is synonymous to timestamp(9):

https://github.com/apache/datafusion/blob/18b2aaa04d956475ddc74fbbde3725370e2d7bde/datafusion/sql/src/planner.rs#L457-L469

Other timestamp types are accessible via `arrow_cast`

https://github.com/apache/datafusion/blob/a2e5330d5cad8516c1524bc86fce32cae324b761/datafusion/sqllogictest/test_files/timestamps.slt#L36-L46

I think we should add SQL-level `timestamp(p)` support as well -- https://github.com/apache/datafusion/pull/13231

**Reply by caicancai** [reply]
Thanks for your answer
